File: Data Structures & Algorithms/longest-consecutive-sequence/submission-0.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        m = {}
        for num in nums:
            m[num] = 1
        
        res = 0
        for num in nums:
            if num-1 not in m:
                i = num
                seq = 0
                while i in m:
                    seq += 1
                    i += 1
                
                res = max(res, seq)
    
        return res

File: Data Structures & Algorithms/duplicate-integer/submission-0.cpp
class Solution {
public:
    bool hasDuplicate(vector<int>& nums) {
        int n = nums.size();
        unordered_map<int, bool> m;

        for(int i=0; i<n; i++){
            if(m[nums[i]]) {
                return true;
            }
            m[nums[i]] = true;
        }

        return false;
    }
};